### ISSUE TITLE
[java] ModifierOrder false positive on "abstract sealed class"

### ISSUE THREAD

**Issue by bmeier-pros** [pmd/pmd#6057]
**Affects PMD Version:** 7.17.0

**Rule:** ModifierOrder https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_codestyle.html#modifierorder
Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_XXX_XXX.html#XXX>

**Description:** False positive with the "sealed" modifier on an abstract class.

**Code Sample demonstrating the issue:**

```java
abstract sealed class Test permits Test1 {}
```

**Expected outcome:**: No error.

PMD reports a violation "ModifierOrder:  Missorted modifiers `abstract sealed`."
but that's wrong according to the JLS (Java 17 version linked for `sealed`): https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.1.1

```text
ClassModifier:
(one of)
[Annotation](https://docs.oracle.com/javase/specs/jls/se17/html/jls-9.html#jls-Annotation) public protected private
abstract static final sealed non-sealed strictfp
```

Note that checkstyle and Google Java format get this right.

**Running PMD through:** *Gradle*

**Comment by bmeier-pros** [comment]
Root cause is that the `JModifier` enum is controlling the order used by the modifiers, and the declaration order is not consistent with the JLS order.
https://github.com/pmd/pmd/blob/main/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JModifier.java
